### ISSUE TITLE
twisted.conch.manhole.Manhole.__init__ passes extra arguments to parent initializer

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2587]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#2587|
|Type|defect|
|Created|2007-04-21 13:35:52Z|

None of Manhole's base classes implements !__init!__ or cares about the namespace parameter.  Calling the base !__init!__ with that parameter causes the class to be uninstantiatable in current python trunk@HEAD (and presumably 2.6, though who knows what the ultimate behavior will be).  Regardless, there's no reason to pass that parameter.

<details><summary>Searchable metadata</summary>

```
trac-id__2587 2587
type__defect defect
reporter__exarkun exarkun
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__conch conch
keywords__ 
time__1177162552000000 1177162552000000
changetime__1177509353000000 1177509353000000
version__None None
owner__ 
cc__therve
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun removed owner|
|-|-|

Here's the fix:

```
Index: twisted/conch/manhole.py
===================================================================
--- twisted/conch/manhole.py    (revision 20019)
+++ twisted/conch/manhole.py    (working copy)
@@ -150,7 +150,7 @@
     namespace = None

     def __init__(self, namespace=None):
-        recvline.HistoricRecvLine.__init__(self, namespace)
+        recvline.HistoricRecvLine.__init__(self)
         if namespace is not None:
             self.namespace = namespace.copy()

```

Without this, about 20 tests fail with python trunk@HEAD.  With it, they pass.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_2"></a>|@therve set owner to @exarkun|
|-|-|

Looks good.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_3"></a>|@therve commented|
|-|-|

Oh, it's just weird that some tests get timeout instead of failing gracefully before the fix.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set status to `closed`|
|-|-|

(In [20066]) Apply patch fixing Manhole.__init__

Author: exarkun
Reviewer: therve
Fixes [#2587](https://github.com/twisted/twisted/issues/2587)

Manhole's initializer calls its parent class's initializer with its namespace
argument.  Manhole is the first class in the hierarchy which cares about this
argument though, and none of its parents have initializers, save object.

Change Manhole's initializer to not pass the parameter up.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
